dts: arm: st: new compatible, for instance st,stm32g0-exti

Desvauxm-st opened the pull request:
by-passed thanks to a new compatible
for instance st,stm32g0-exti added to the matching targets: G0/U5/L5/MP1:

Desvauxm-st committed:
driver: remove #ifdef CONFIG_SOC_SERIES_STM32xx

remove all #ifdef CONFIG_SOC_SERIES_STM32xx
before to add a st,stm32g0-exti compatible
added to the matching targets:
G0/U5/L5/MP1:

Signed-off-by: Marc Desvaux <marc.desvaux-ext@st.com>

erwango reviewed (changes requested):
Please fix commit title.

Also please split into commits:
- dts
- driver changes

Desvauxm-st commented on dts/arm/st/l5/stm32l5.dtsi:
```diff
@@ -134,7 +134,7 @@
 
 		exti: interrupt-controller@4000f400 {
-			compatible = "st,stm32-exti";
+			compatible = "st,stm32g0-exti", "st,stm32-exti";
```
I don't know how to do that

Desvauxm-st replied:
add the new  file st,stm32g0-exti.yaml into dts/bindings/interrupt-controller

Desvauxm-st committed:
dts: arm: st: Introduce a st,stm32g0-exti compatible

Introduce a st,stm32g0-exti compatible
 added to the matching targets:
G0/U5/L5/MP1:

Signed-off-by: Marc Desvaux <marc.desvaux-ext@st.com>

Desvauxm-st committed:
Merge branch 'main' into new_compatible_instance_st_stm32g0-exti

erwango commented:
Please rebase on top of latest main branch and remove "merge commit".

Desvauxm-st commented:
see #55844
